js/views/message_detail_view.js

```diff
@@ -98,14 +132,21 @@
             }));
             this.view.$el.prependTo(this.$('.message-container'));
 
+            this.grouped = _.groupBy(this.model.get('errors'), 'number');
             if (this.model.isOutgoing()) {
-                this.conversation.contactCollection.reject(function(c) {
+                var contacts = this.conversation.contactCollection.reject(function(c) {
                     return c.isMe();
-                }).forEach(this.renderContact.bind(this));
+                });
+
+                _.sortBy(contacts, function(c) {
+                    var prefix = this.grouped[c.id] ? '0' : '1';
```

Comment:
I see. First time through, I totally missed that there was a concatenation happening in the return statement.